src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -569,12 +569,12 @@ bool primitive_inst::update_impl() {
         }
         if (!cached_impl) {
             if (_dynamic_impl) {
-                auto use_async_compilation = [&]() {
+                auto use_async_compilation = [&updated_params_no_dyn_pad]() {
                     GPU_DEBUG_GET_INSTANCE(debug_config);
                     GPU_DEBUG_IF(debug_config->disable_async_compilation) {
                         return false;
                     }
-                    return true;
+                    return updated_params_no_dyn_pad.desc->type_string().compare("fully_connected") == 0;
```

Comment:
Why comparing strings?
```
updated_params_no_dyn_pad.is_type<fully_connected>()
```
it can also be a member of primitive_inst
```cpp
bool primitive_inst::use_async_compilation() const {
    return _node->is_type<fully_connected>();
}
```

Reply:
Oh, I'll update it as you guided. Thank you!